src/main/java/io/vertx/core/http/impl/Http2ServerStream.java

```diff
@@ -60,26 +62,35 @@ class Http2ServerStream extends VertxHttp2Stream<Http2ServerConnection> {
     this.headers = null;
     this.method = method;
     this.uri = uri;
+    this.scheme = null;
     this.host = null;
+    this.authorityString = null;
     this.authority = null;
     this.tracingPolicy = tracingPolicy;
     this.halfClosedRemote = halfClosedRemote;
   }
 
-  Http2ServerStream(Http2ServerConnection conn, ContextInternal context, Http2Headers headers, String serverOrigin, TracingPolicy tracingPolicy, boolean halfClosedRemote) {
+  Http2ServerStream(Http2ServerConnection conn, ContextInternal context, Http2Headers headers, TracingPolicy tracingPolicy, boolean halfClosedRemote) {
     super(conn, context);
 
-    String host = headers.get(":authority") != null ? headers.get(":authority").toString() : null;
-    if (host == null) {
-      int idx = serverOrigin.indexOf("://");
-      host = serverOrigin.substring(idx + 3);
+    CharSequence schemeHeader = headers.getAndRemove(":scheme");
```

Comment:
I know this commit was made a PoC to quantify the improvement, but I'm adding this notes to not forget...
All of these can be made faster by using the proper `AsciiString` values' on the Netty PseudoHeader's enums: differently, by using `String` literals, both the hash code computation and equals is going to be performed char-by-char and never cached.